extensions/chooser/HSChooser.m

```diff
@@ -393,7 +394,19 @@ - (void)tableView:(NSTableView *)tableView didClickedRow:(NSInteger)row {
             [skin pushNSObject:choice];
             [skin protectedCallAndError:@"hs.chooser:completionCallback" nargs:1 nresults:0];
         }
-        
+
+        _lua_stackguard_exit(skin.L);
+    } else if (self.enableDefaultForQuery != NO) {
```

Comment:
This should also check that `self.completionCallbackRef` is neither `LUA_NOREF` or `LUA_REFNIL`